File: examples/docs-code-snippets/src/main/scala/examples/scala/application/AwaitEvent.scala
package examples.scala.application

import com.ing.baker.runtime.scaladsl.{Baker, EventInstance}
import examples.scala.events.OrderPlaced

import scala.concurrent.ExecutionContext.Implicits.global
import scala.concurrent.duration.DurationInt

class AwaitEvent(val baker: Baker) {

  def example(recipeInstanceId: String, orderPlaced: OrderPlaced): Unit = {
    val eventInstance = EventInstance.unsafeFrom(orderPlaced)
    for {
      _ <- baker.fireSensoryEventAndAwaitReceived(recipeInstanceId, eventInstance)
      _ <- baker.awaitEvent(recipeInstanceId, "ExpectedEvent", timeout = 5.seconds)
    } yield ()
  }
}

File: examples/docs-code-snippets/src/main/java/examples/java/application/AwaitEvent.java
package examples.java.application;

import com.ing.baker.runtime.javadsl.Baker;
import com.ing.baker.runtime.javadsl.EventInstance;
import examples.java.events.OrderPlaced;

import java.time.Duration;

public class AwaitEvent {

    private final Baker baker;

    public AwaitEvent(Baker baker) {
        this.baker = baker;
    }

    public void example(String recipeInstanceId, OrderPlaced orderPlaced) {
        var eventInstance = EventInstance.from(orderPlaced);
        baker.fireSensoryEventAndAwaitReceived(recipeInstanceId, eventInstance).join();
        baker.awaitEvent(recipeInstanceId, "ExpectedEvent", Duration.ofSeconds(5)).join();
    }
}
